core/ast.h

```diff
@@ -126,7 +128,8 @@ struct AST {
     virtual ~AST(void) {}
 };
 
-typedef std::vector<AST *> ASTs;
+#include <list>
+typedef std::list<AST *> ASTs;
```

Comment:
I'm skeptical that `std::list` will improve anything. It requires more mallocs and consumes more memory than `std::vector`.